tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -2202,7 +2202,8 @@ void CGMSHLSLRuntime::AddHLSLFunctionInfo(Function *F, const FunctionDecl *FD) {
           rayShaderHaveErrors = true;
         }
         if (ArgNo < 2) {
-          if (!IsHLSLNumericUserDefinedType(parmDecl->getType())) {
+          if (!(IsHLSLNumericUserDefinedType(parmDecl->getType()) ||
+                IsHLSLBuiltinRayAttributeStruct(parmDecl->getType()))) {
```

Comment:
`BuiltInTriangleIntersectionAttributes` should be allowed here, but not `RaytracingAccelerationStructure`, since that's an opaque resource type.  So this code shouldn't be treating them the same.